app/controllers/users_controller.rb

```diff
@@ -1740,6 +1740,68 @@ def bookmarks
     end
   end
 
+  USER_MENU_BOOKMARKS_LIST_LIMIT = 20
+  def user_menu_bookmarks
+    user = fetch_user_from_params
+    guardian.ensure_can_see_notifications!(user)
+    user_guardian = Guardian.new(user)
+
+    if params[:limit]
+      limit = params[:limit].to_i
+      limit = limit.clamp(1..USER_MENU_BOOKMARKS_LIST_LIMIT)
+    else
+      limit = USER_MENU_BOOKMARKS_LIST_LIMIT
+    end
+
+    reminder_notifications = user
+      .notifications
+      .visible
+      .includes(:topic)
+      .where(read: false, notification_type: Notification.types[:bookmark_reminder])
+      .limit(limit)
+      .to_a
+
+    if reminder_notifications.size < limit
+      exclude_bookmark_ids = reminder_notifications
+        .filter_map { |notification| notification.data_hash[:bookmark_id] }
+
+      bookmark_list = UserBookmarkList.new(
+        user: user,
+        guardian: user_guardian,
+        params: {
+          per_page: limit - reminder_notifications.size
+        }
+      )
+      bookmark_list.load do |query|
+        if exclude_bookmark_ids.present?
+          query.where("bookmarks.id NOT IN (?)", exclude_bookmark_ids)
+        end
+      end
+    end
+
+    if reminder_notifications.present?
+      serialized_notifications = ActiveModel::ArraySerializer.new(
+        reminder_notifications,
+        each_serializer: NotificationSerializer,
+        scope: user_guardian
+      )
+    end
+
+    if bookmark_list
+      serialized_bookmarks = serialize_data(
+        bookmark_list,
+        UserBookmarkListSerializer,
+        scope: user_guardian,
+        root: false
+      )[:bookmarks]
+    end
+
+    render json: {
+      notifications: serialized_notifications || [],
+      bookmarks: serialized_bookmarks || []
+    }, status: 200
```

Comment:
```suggestion
    }
```

200 is already the default.